orc8r/cloud/go/services/analytics/calculations/calculations.go

```diff
@@ -131,26 +147,23 @@ func CombineLabels(l1, l2 map[string]string) map[string]string {
 	return retLabels
 }
 
-//RegisterResults exports the metrics to prometheus
-func RegisterResults(calc CalculationParams, results []*protos.CalculationResult) {
+// RegisterResults exports the metrics to prometheus
+func RegisterResults(calc *CalculationParams, results []*protos.CalculationResult) {
 	for _, res := range results {
+		if calc.RegisteredGauge == nil {
+			glog.Errorf("Attempting to register with %s non existent gauge ", res.MetricName)
+			continue
+		}
 		if !CheckLabelsMatch(calc.ExpectedGaugeLabels, res.Labels) {
 			glog.Errorf("Unmatched labels in Calculation. Expected: %s, Received: %s", calc.ExpectedGaugeLabels, printLabels(res.Labels))
 			continue
 		}
-		if calc.MetricConfig != nil {
-			metricConfig, ok := calc.MetricConfig[res.MetricName]
-			if ok && metricConfig.Register == false {
-				glog.V(2).Infof("%s result registration skipped due to configuration", calc.Name)
-				return
-			}
-		}
 		calc.RegisteredGauge.With(res.Labels).Set(res.Value)
 		glog.V(10).Infof("Set metric %s{%s} value: %f\n", res.MetricName, printLabels(res.Labels), res.Value)
```

Comment:
Instead of adding a custom formatter here for protos.CalculationResult, define the String() method on the type in the protos package

Reply:
I have only this single instance where I print it, it is more an overhead to add a method here. If you feel strongly, I can add a method. LMK.

Reply:
Not the end of the world. But preference to just default to using String() when you find yourself hand-coding a formatting string

The main reason I care is by-hand debugging -- if I need to print the contents of that proto, the first thing I'm gonna do is define the String method. So may as well define it now